examples/go/tagging/handler/handler.go

```diff
@@ -31,140 +32,145 @@ type vcConfig struct {
+	moRef, err := parseEventMoRef(req.Body)
 	if err != nil {
+		log.Fatalf("retrieve managed reference object failed: %v", err)
 		return handler.Response{
 			Body:       nil,
 			StatusCode: http.StatusBadRequest,
 		}, err
 	}
 
-	if event.Event.Vm.Vm.Value == "" || event.Event.Vm.Vm.Type == "" {
+	log.Println("tag VM moRef with tagID (URN)")
+	err = client.MoTag(ctx, moRef, cfg.Tag.Urn)
+	if err != nil {
+		log.Fatalf("tagging managed reference object failed: %v", err)
 		return handler.Response{
 			Body:       nil,
 			StatusCode: http.StatusBadRequest,
-		}, errors.New("managed Reference Object information must not be empty")
+		}, err
 	}
 
-	// Fill information in the request into a govmomi type.
-	var moRef types.ManagedObjectReference
-	moRef.Value = event.Event.Vm.Vm.Value
-	moRef.Type = event.Event.Vm.Vm.Type
-
-	// Connect to vcenter with govmomi client
-	log.Println("connect to vCenter")
-	clients := connect(ctx, vcc.VCenter)
-	log.Println("connection successful")
-
-	// tag the VM moRef with tagID (URN)
-	clients.tagVM(ctx, moRef, vcc.Tag.Urn)
-
 	// Respond with a message that tagging of vm was complete
 	message := fmt.Sprintf(
 		"%v was tagged with %v",
 		moRef.Value,
-		vcc.Tag.Urn,
+		cfg.Tag.Urn,
 	)
 	log.Println(message)
 
-	err = clients.Govmomi.Logout(ctx)
-	if err != nil {
-		log.Printf("unable to log out of govmomi due to %v", err)
-	}
-	err = clients.Rest.Logout(ctx)
-	if err != nil {
-		log.Printf("unable to log out of rest due to %v", err)
-	}
-
 	return handler.Response{
 		Body:       []byte(message),
 		StatusCode: http.StatusOK,
 	}, nil
 }
 
-func loadTomlCfg() (vcConfig, error) {
-	secretPath := "/var/openfaas/secrets/vcconfig"
-	secret, err := toml.LoadFile(secretPath)
+// vsConnect connects to vSphere govmomi API using information from vcconfig.toml.
+func vsConnect() error {
+	var (
+		err        error
+		configPath = "/var/openfaas/secrets/vcconfig"
+	)
+
+	// Retrieve vCenter and tag info from the vcconfig file.
+	log.Println("load vcconfig.toml")
+	cfg, err = loadTomlCfg(configPath)
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("vcconfig.toml loading failed: %w", err)
 	}
 
-	cfg := vcConfig{}
-	secret.Unmarshal(&cfg)
+	u := url.URL{
+		Scheme: "https",
+		Host:   cfg.VCenter.Server,
+		Path:   "sdk",
+	}
+	u.User = url.UserPassword(cfg.VCenter.User, cfg.VCenter.Password)
 
-	return cfg, nil
-}
+	// Currently, VEBA does not support signed certificates.
+	insecure := true
 
-// Clients needed to access vSphere APIs
-type Clients struct {
-	Govmomi *govmomi.Client
-	Rest    *rest.Client
+	log.Println("connect to vSphere")
+	client, err = vsClient.New(ctx, u, insecure)
+	if err != nil {
+		return fmt.Errorf("connection to vSphere API failed: %w", err)
+	}
+
+	return nil
 }
 
-// Connect saves connection information to clients.
-func connect(ctx context.Context, vca vCenterAuth) *Clients {
-	var c Clients
+func loadTomlCfg(path string) (vcConfig, error) {
+	cfg := vcConfig{}
```

Comment:
You redeclare `cfg` here (either make local var or global - but not both) and names are mixed (global, local and return var), ie. not consistent.

Reply:
Oops--I'll try to make `cfg` a local var. As for names, I'll try not to use `cfg` instead of `config`, regardless of the type of variable (global, local, returned).